source/MaterialXView/Editor.cpp

```diff
@@ -98,6 +98,19 @@ void PropertyEditor::addItemToForm(const mx::UIPropertyItem& item, const std::st
         form.addGroup(group);
     }
 
+    // Find out if the uniform is editable. Some
+    // inputs may be optimized out during compilation.
```

Comment:
Instead of "editable" or not would it make sense to not create the variable at all if it's not used by the shader? Maybe also add a TODO comment about fixing this properly on shader gen side instead later.

Reply:
It can still be used in the shader but just as a constant which has been optimized out by the compiler so we can't edit it.

Reply:
Just looked at it again. It is properly good to not show non-editable things.